packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx

```diff
@@ -1,94 +1,323 @@
+    const truncatedReleaseDate = moment(releaseDateField.value).startOf('day')
+
+    const today = moment().startOf('day')
+
+    if (moment(truncatedReleaseDate).isBefore(today)) {
+      setTimePeriod(TimePeriodType.PAST)
+    } else if (moment(truncatedReleaseDate).isAfter(today)) {
+      setTimePeriod(TimePeriodType.FUTURE)
+      setReleaseDateHour('12:00')
+      setReleaseDateMeridian(ReleaseDateMeridian.AM)
+    } else {
+      setTimePeriod(TimePeriodType.PRESENT)
+    }
+  }, [releaseDateField.value, setReleaseDateHour, setReleaseDateMeridian])
+
+  return (
+    <>
+      <RadioButtonGroup
+        {...releaseDateTypeField}
+        className={styles.radioGroup}
+        defaultValue={releaseDateTypeField.value ?? ReleaseDateType.RELEASE_NOW}
+      >
+        <ModalRadioItem
+          value={ReleaseDateType.RELEASE_NOW}
+          label='Release Immediately'
+        />
+        <ModalRadioItem
+          value={ReleaseDateType.SCHEDULED_RELEASE}
+          label='Select a release date'
+        />
+      </RadioButtonGroup>
+      {releaseDateTypeField?.value === ReleaseDateType.SCHEDULED_RELEASE && (
+        <div
+          className={cn(
+            styles.dropdownRow,
+            layoutStyles.row,
+            layoutStyles.gap2,
+            styles.releaseDateTimePicker
+          )}
+        >
           <div className={styles.datePicker}>
-            <DatePickerField name={RELEASE_DATE} label={messages.title} />
+            <DatePickerField
+              isScheduledRelease={true}
+              name={RELEASE_DATE}
+              label={messages.title}
+              shouldFocus={
+                releaseDateTypeField.value === ReleaseDateType.SCHEDULED_RELEASE
+              }
+            />
           </div>
-        </>
-      }
-      renderValue={renderValue}
+          {timePeriod !== TimePeriodType.PAST && (
+            <>
+              <HarmonyTextField
+                name={RELEASE_DATE_HOUR}
+                label={'Time'}
+                placeholder={'12:00'}
+                hideLabel={false}
+                inputRootClassName={styles.hourInput}
+                transformBlurValue={(value) => {
+                  if (value.includes(':')) {
+                    return value
+                  }
+                  // add :00 if it's missing
+                  const number = parseInt(value, 10)
+                  if (!isNaN(number) && number >= 1 && number <= 12) {
+                    return `${number}:00`
+                  }
+                  return value
+                }}
+              />
+              <SelectMeridianField />
+            </>
+          )}
+        </div>
+      )}
+      <ModalContent className={styles.releaseDateHint}>
+        <HelpCallout
+          icon={<IconInfo />}
+          content={messages.callout(timePeriod)}
+        />
+      </ModalContent>
+    </>
+  )
+}
+
+export const SelectMeridianField = () => {
+  return (
+    <DropdownField
+      aria-label={'label'}
```

Comment:
this should be `release date meridian` or something, right?

Reply:
just removed that label, not sure what it does tbh